src/modeci_mdf/interfaces/pytorch/importer.py

```diff
@@ -576,6 +580,290 @@ def pytorch_to_mdf(
+def get_argument_id(function_name):
+    func_vs_arglist = {
+        "onnx::Gemm": ["A", "B", "C"],
+        "onnx::Relu": ["X"],
+        "onnx::Reshape": ["data", "shape"],
+    }
+    if function_name in func_vs_arglist:
+        if not hasattr(get_argument_id, "index"):
+            get_argument_id.index = {}
+        if function_name not in get_argument_id.index:
+            get_argument_id.index[function_name] = 0
+        else:
+            get_argument_id.index[function_name] = (
+                get_argument_id.index[function_name] + 1
+            ) % len(func_vs_arglist[function_name])
+        return func_vs_arglist[function_name][get_argument_id.index[function_name]]
+    else:
+        return None
+
+
+# https://pytorch.org/docs/stable/fx.html#the-interpreter-pattern
+class ShapeProp:
+    """
+    Shape propagation. This class takes a `GraphModule`.
+    Then, its `propagate` method executes the `GraphModule`
+    node-by-node with the given arguments.
+    """
+
+    def __init__(self, mod, portmapper_obj):
+        self.mod = mod
+        self.graph = torch.fx.symbolic_trace(mod).graph
+        self.modules = dict(self.mod.named_modules())
+        self.pm = portmapper_obj
+        # Due to some reason info about weigths and biases of each node is not present in any of its attributes
+        # So, this is a hack to get the assosciated weights and biases to a particular layer
+        # eg if node name is fc1 and fc1 is present as a substring in the fx_dict keys then we simply map
+        # the node name with the keys matched eg - fc1:[fc1.weight,fc1.bias], view[] ,.....
+        self.node_wbs = {
+            node.name: [key for key in self.pm.fx_dict.keys() if node.name in key]
+            for node in self.graph.nodes
+        }
+
+    def propagate(self, *args):
+        args_iter = iter(args)
+        env: Dict[str, fx_Node] = {}
+        prev_in_shape = args[0].shape
+        prev_in_type = args[0].dtype
+        mdf_model = Model(id="sample_model")
+        mdf_graph = Graph(id="sample_graph")
+        mdf_model.graphs.append(mdf_graph)
+
+        def load_arg(a):
+            return torch.fx.graph.map_arg(a, lambda n: env[n.name])
+
+        for node in self.graph.nodes:
+            para_gen = parameter_id_generator()  # initiate generator for parameter id
+
+            mdf_node = Node(id=node.name)
+            shape, type = parse_shape_type(prev_in_shape, prev_in_type)
+
+            argument = {}
+            for ip in node.all_input_nodes:
+                mdf_node.input_ports.append(
+                    InputPort(id=ip.name, shape=shape, type=type)
+                )
+
+            func = None
+            arg_values = list(node.args) + self.node_wbs[node.name]
+
+            if node.op == "placeholder":
+                result = next(args_iter)
+
+            elif node.op == "call_function":
+                func = "onnx::Relu"
```

Comment:
I think we need a more general way to identify the appropriate ONNX function for the torch op. I think this might be something that the torch and ONNX folks are working on as well, see [here](https://github.com/pytorch/pytorch/blob/af440c427bffddd7edc2db800a0c34ae8f1107d0/torch/onnx/_internal/fx/passes/fx_to_onnxscript.py#L351). It looks like they are developing a converter between torch fx IR to onnxscript.  I didn't know that ONNX now has a thing called ONNX script. Allowing for composing ONNX ops in Python. Might be useful for re-writing the execution engine in fact.

Anyway, I think this issue needs to be fixed before merging.

Reply:
I did go through their implementation and they seem to get the appropriate function from [here](https://github.com/pytorch/pytorch/blob/af440c427bffddd7edc2db800a0c34ae8f1107d0/torch/onnx/_internal/fx/function_dispatcher.py#L188). But cannot figure out how to use a key in  **_OP_OVERLOAD_TO_EXPORTER_KEY_TABLE** dictionary to get the value.